src/user/user.service.ts

```diff
@@ -114,14 +114,18 @@ export class UserService {
 
   async createUser(dto: CreateUserDto): Promise<UserEntity> {
     const { avatar_id, ...userDetails } = dto;
-    const avatar = await this.imageRepository.findOneBy({
-      id: avatar_id,
-    });
-    if (!avatar)
-      throw new HttpException(
-        `Avatar with id: ${avatar_id} does not exist`,
-        HttpStatus.NOT_FOUND,
-      );
+    let avatar: ImageEntity = null;
+
+    if (avatar_id) {
+      avatar = await this.imageRepository.findOneBy({
+        id: avatar_id,
+      });
+      if (!avatar)
+        throw new HttpException(
+          `Avatar with id: ${avatar_id} does not exist`,
+          HttpStatus.NOT_FOUND,
```

Comment:
If I'm getting this right - if an `avatar_id` is provided but not found in the database, the user creation will fail and return a `404 Not Found` response? This looks like a `400 Bad Request` to me, since the request has been made with an invalid avatar id.

Reply:
Ah, true. Fixing